### DISCUSSION TITLE
Moment.js Locale Override Mystery - AM/PM Parsing Gone Wrong!"

### DISCUSSION THREAD

**Discussion by shakeelkoper** [moment/moment#6414] [Q&A] [answered]
**Describe the bug**
The `moment().locale()` function is not able to override the locale from 'ja' to 'en' for a specific instance. As a result, it's not able to correctly parse the "AM/PM" part of a UTC date-time string. ex: '5/11/2024 3:00:31 PM'

**To Reproduce**
1. Set the global locale to 'ja' using `moment.locale('ja')`.
2. Try to parse a date-time string in English format with the "AM/PM" part, using `moment().locale('en')`.
ex:`moment("5/11/2024 3:00:31 PM", "M/D/YYYY h:mm:ss a").locale('en').utcOffset(0, true).unix()`
3. The "AM/PM" part is not parsed correctly, as indicated by `unusedInput:["PM"]` and `unusedTokens: ["a"]`.

**Expected behavior**
The `moment().locale('en')` function should override the global locale for the specific instance, allowing the "AM/PM" part of the date-time string to be parsed correctly.
`1715439631` -> `Saturday, 11 May 2024 15:00:31`

**Wrong behavior**
`1715396431` -> `Saturday, 11 May 2024 03:00:31`

**Desktop (please complete the following information):**
 - OS: macOS 14.4.1 (23E224)
 - Node.js version: v16.20.2
 - Moment.js version: 2.30.1

**Moment-specific environment**

* The time zone setting of the machine the code is running on: [e.g. JST]
* The time and date at which the code was run: [e.g. '5/10/2024 8:58:17 PM']
* Other libraries in use: moment-duration-format 2.3.2, moment-timezone 0.5.34

Please run the following code in your environment and include the output:
```javascript
console.log((new Date()).toString()) //'Thu May 16 2024 01:28:31 GMT+0530 (India Standard Time)'
console.log((new Date()).toLocaleString()) //'16/05/2024, 01:28:47'
console.log((new Date()).getTimezoneOffset()) //-330
console.log(process.version)
console.log(moment.version) //'2.30.1'

**Comment by shivansh84ya** [comment]
Hi @shakeelkoper ,
I would like to work on this issue. Could you please assign it to me?
Thanks!

**Comment by shakeelkoper** [comment]
@shivansh84ya I am not able to change the assignee
<img width="1251" alt="image" src="https://github.com/moment/moment/assets/18098743/be1e33e0-290d-4f50-a9ce-1ce4f7d6ce12">

**Comment by shivansh84ya** [comment]
should i start working on it??

**Comment by shakeelkoper** [comment]
yes please

**Comment by gilmoreorless** [comment, marked as answer]
> `moment("5/11/2024 3:00:31 PM", "M/D/YYYY h:mm:ss a").locale('en').utcOffset(0, true).unix()`

In your code example, you're only calling `.locale('en')` _after_ the parsing has already happened. That snippet is not trying to parse the string in `en` locale; it's trying to parse the string in the global locale _then_ use the `en` locale afterwards.

**Comment by shakeelkoper** [comment]
can you please provide the sample code

**Comment by gilmoreorless** [comment]
There are already examples in the [parsing documentation](https://momentjs.com/docs/#/parsing/):

> As of version **2.0.0**, a locale key can be passed as the third parameter to `moment()` and `moment.utc()`.
>
> ```js
> moment('2012 juillet', 'YYYY MMM', 'fr');
> moment('2012 July',    'YYYY MMM', 'en');
> moment('2012 July',    'YYYY MMM', ['qj', 'en']);
> ```
